src/main/java/org/broadinstitute/hellbender/tools/spark/utils/HopscotchHashSet.java

```diff
@@ -0,0 +1,542 @@
+ * This is usually a little slower than the JDK's HashSet (neglecting GC time), but it uses much less memory.
+ * It's probably a nice choice for very large collections.
+ * You can make it pretty much equally fast as HashSet by replacing the prime-number table sizing with power-of-2 table
+ * sizing, but then it'll behave just as badly as HashSet given crappy hashCode implementations.
+ */
+@DefaultSerializer(HopscotchHashSet.Serializer.class)
+public final class HopscotchHashSet<T> extends AbstractSet<T> {
+    // For power-of-2 table sizes add this line
+    //private static final int maxCapacity = Integer.MAX_VALUE/2 + 1;
+
+    private int capacity;
+    // unused buckets contain null.  (this data structure does not support null entries.)
+    // if the bucket is unused, the corresponding status byte is irrelevant, but is always set to 0.
+    private int size;
+    private T[] buckets;
+    // format of the status bytes:
+    // high bit set indicates that the bucket contains a "chain head" (i.e., an entry that naturally belongs in the
+    // corresponding bucket).  high bit not set indicates a "squatter" (i.e., an entry that got placed here through the
+    // collision resolution methodology).  we use Byte.MIN_VALUE (i.e., 0x80) to pick off this bit.
+    // low 7 bits give the (unsigned) offset from the current entry to the next entry in the collision resolution chain.
+    // if the low 7 bits are 0, then we'd be pointing at ourselves, which is nonsense, so that particular value marks
+    // "end of chain" instead.  we use Byte.MAX_VALUE (i.e., 0x7f) to pick off these bits.
+    private byte[] status;
+
+    @VisibleForTesting static final double LOAD_FACTOR = .85;
+
+    // largest prime numbers less than each half power of 2 from 2^8 to 2^31
+    @VisibleForTesting static final int[] legalSizes = {
+            251, 359, 509, 719, 1021, 1447, 2039, 2887, 4093, 5791, 8191, 11579, 16381, 23167, 32749, 46337, 65521,
+            92681, 131071, 185363, 262139, 370723, 524287, 741431, 1048573, 1482907, 2097143, 2965819, 4194301, 5931641,
+            8388593, 11863279, 16777213, 23726561, 33554393, 47453111, 67108859, 94906249, 134217689, 189812507,
+            268435399, 379625047, 536870909, 759250111, 1073741789, 1518500213, 2147483647
+    };
+    private static final int SPREADER = 241;
+
+    @SuppressWarnings("unchecked")
+    public HopscotchHashSet( final int capacity ) {
+        this.capacity = computeCapacity(capacity);
+        this.size = 0;
+        // Not unsafe, because the remainder of the API allows only elements known to be T's to be assigned to buckets.
+        this.buckets = (T[])new Object[this.capacity];
+        this.status = new byte[this.capacity];
+    }
+
+    public HopscotchHashSet( final Collection<T> collection ) {
+        this(collection.size());
+        addAll(collection);
+    }
+
+    public HopscotchHashSet( final Collection<T> collection, final float sizeInflator ) {
```

Comment:
It's for when you're building from a collection, but know that you'll be adding more.  It's fairly silly.  I'll get rid of it.